src/model/presentation/xml/SVGFile.java

```diff
@@ -93,9 +120,21 @@ public void addShapes(List<Shape> shapes) {
 		if (shapes != null)
 			this.shapes.addAll(shapes);
 	}
+	
+	@Override
+	public void insert(Presentation svg) {
+		if(this==svg){
+			try {
+				throw new Exception();
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+			this.shapes.add(svg);
+	}
 
 	@Override
-	public void display() {
+	public String display() {
 		this.xmlcontent = "";
```

Comment:
I do agree, but it was in the interface.
As it is in the interface, it must be generic and not something with "xml" in it.
Why not "CreateDrawing" or get "GetDisplay" and the display itself just return the xml in that case, but adding shapes would update the xml attribute.
@starsasumi ?